recipe/meta.yaml

```
@@ -13,11 +13,15 @@ source:
     - 0001-Use-CMAKE_SYSTEM_PROCESSOR-instead-of-CMAKE_LIBRARY_.patch
 
 build:
-  number: 0
+  number: 1
   string: h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}{{ build_ext }}
   ignore_run_exports:
     - jemalloc        # [(not osx) and (folly_build_ext is not undefined and folly_build_ext == "jemalloc")]
     - jemalloc-local  # [osx and (folly_build_ext is not undefined and folly_build_ext == "jemalloc")]
+  run_exports:
+    # Since folly is ABI-unstable, the exact same version must be used at compile time and runtime.
+    # See: https://github.com/facebook/folly/#build-notes
+    - {{ pin_subpackage('folly', exact=True) }}
```

Rebuilds of the same folly version should be ABI stable (we're not changing any folly code).
```suggestion
    - {{ pin_subpackage('folly', max_pin='x.x.x.x') }}
```

---

Before integrating this change, I want to understand how this is different from using `exact=True`. I unfortunately can't find any information of the usage of `exact=True`, but doesn't exact=True` specifies to use the exact same version? Or does it specify something else (e.g. something more granular, like using the exact same build)?

---

`exact=True` pins down to the build number and hash. This is very strict and normally not what you want.

---

OK thanks.